guvnor-ala/guvnor-ala-build-maven/src/main/java/org/guvnor/ala/build/maven/executor/MavenProjectConfigExecutor.java

```diff
@@ -0,0 +1,119 @@
+    }
+
+    @Override
+    public Optional<ProjectConfig> apply( final Source source,
+                                          final MavenProjectConfig mavenProjectConfig ) {
+        final InputStream pomStream = Files.newInputStream( source.getPath().resolve( mavenProjectConfig.getProjectDir() ).resolve( "pom.xml" ) );
+        final MavenProject project = MavenProjectLoader.parseMavenPom( pomStream );
+
+        final Collection<PlugIn> buildPlugins = extractPlugins( project );
+
+        final String expectedBinary = project.getArtifact().getArtifactId() + "-" + project.getArtifact().getVersion() + "." + project.getArtifact().getType();
+        final org.guvnor.ala.build.maven.model.MavenProject mavenProject = new MavenProjectImpl( project.getId(),
+                                                                                                 project.getArtifact().getType(),
+                                                                                                 project.getName(),
+                                                                                                 expectedBinary,
+                                                                                                 source.getPath(),
+                                                                                                 source.getPath().resolve( mavenProjectConfig.getProjectDir() ),
+                                                                                                 source.getPath().resolve( "target" ).resolve( expectedBinary ).toAbsolutePath(),
+                                                                                                 buildPlugins );
+
+        sourceRegistry.registerProject( source, mavenProject );
+
+        return Optional.of( mavenProject );
+    }
+
+    private Collection<PlugIn> extractPlugins( final MavenProject project ) {
+        final Collection<PlugIn> result = new ArrayList<>( project.getBuildPlugins().size() );
+        for ( org.apache.maven.model.Plugin plugin : project.getBuildPlugins() ) {
+            final Map<String, Object> config = extractConfig( plugin.getConfiguration() );
+            result.add( new PlugIn() {
+                @Override
+                public String getId() {
+                    return plugin.getKey();
+                }
+
+                @Override
+                public Map<String, ?> getConfiguration() {
+                    return config;
+                }
+            } );
+        }
+        return result;
+    }
+
+    private Map<String, Object> extractConfig( final Object configuration ) {
+        if ( configuration instanceof Xpp3Dom ) {
+            final Map<String, Object> result = new HashMap<>();
+            extractConfig( result, (Xpp3Dom) configuration );
+            if ( result.containsKey( "configuration" ) ) {
+                try {
+                    return (Map<String, Object>) result.get( "configuration" );
```

Comment:
What exception can happen hear and why is it okay to return an empty map instead? A brief comment here and/or catching a more specific exception would make this easier to follow.